core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java

```diff
@@ -224,23 +217,111 @@ public T unmarshall(JsonUnmarshallerContext context,
 
     public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
```

Comment:
dumb question: What is the reason for the branching here? i.e. why isn't this just a call to `unmarshallResponse`? I assume `unmarshallFromJson` is the fast-path here?

Reply:
Yes, that's the idea. Most of the complications here are to support restJson, RPC protocols are a lot simpler and do not need any of that, so we attempt to separate one from the other to avoid paying the cost for all.